sim/src/main/makefrag/firesim/Makefrag

```diff
@@ -30,22 +30,57 @@ VERILOG := $(GENERATED_DIR)/FPGATop.v
 HEADER  := $(GENERATED_DIR)/$(DESIGN)-const.h
 
 ifdef FIRESIM_STANDALONE
-	firesim_sbt_project := {file:${chipyard_dir}}firechip
+	firesim_sbt_project ?= {file:${chipyard_dir}}firechip
 
 	lookup_scala_srcs = $(shell find -L $(1)/ -name target -prune -o -iname "[!.]*.scala" -print 2> /dev/null)
-	SOURCE_DIRS = $(chipyard_dir)/generators $(firesim_base_dir)
-	SCALA_SOURCES = $(call lookup_scala_srcs,$(SOURCE_DIRS))
+	SOURCE_DIRS ?= $(chipyard_dir)/generators $(firesim_base_dir)
+	SCALA_SOURCES ?= $(call lookup_scala_srcs,$(SOURCE_DIRS))
 else
-	firesim_sbt_project := firechip
+	firesim_sbt_project ?= firechip
 endif
 
-$(FIRRTL_FILE) $(ANNO_FILE): $(SCALA_SOURCES) $(FIRRTL_JAR) $(SCALA_BUILDTOOL_DEPS)
-	mkdir -p $(@D)
-	$(call run_scala_main,$(firesim_sbt_project),chipyard.Generator,\
+BOOTROM_FILES = bootrom.rv32.img bootrom.rv64.img
+CHIPYARD_ROMDIR ?= $(chipyard_dir)/generators/testchipip/bootrom
+# pattern rule to handle copying bootrom images into GENERATED_DIR so that they are available for
+# - distributed elaboration on a build host
+# - dromajo co-sim
+BOOTROM_TARGETS = $(addprefix $(GENERATED_DIR)/, $(BOOTROM_FILES))
+$(BOOTROM_TARGETS): $(GENERATED_DIR)/bootrom.%.img: $(CHIPYARD_ROMDIR)/bootrom.%.img | $(GENERATED_DIR)
```

Comment:
Im not opposed to this... but if we want to dedup even more we could make the `none` Simulator option in `GenerateSimFiles` (in Chipyard) just copy the bootrom... I think that should work with the FPGA prototyping stuff.

Reply:
I'll look at that and if I'm able to move it in there, I'll move this target into my private TARGET_PROJECT_MAKEFRAG (which doesn't use the CY generator)

Reply:
Hey @abejgonzalez, when I took the copying out of the makefiles and sanity checked it, I run into this error:
```
cd /home/centos/chipyard_oss_210430 && java -Xmx8G -Xss8M -XX:MaxPermSize=256M -Djava.io.tmpdir=/home/centos/chipyard_oss_210430/.java_tmp -cp /home/centos/chipyard_oss_210430/sims/firesim/sim/generated-src/firechip.jar chipyard.Generator --target-dir /home/centos/chipyard_oss_210430/sims/firesim/sim/generated-src/f1/FireSim-FireSimRocketConfig-BaseF1Config --name firesim.firesim.FireSim.FireSimRocketConfig --top-module firesim.firesim.FireSim --legacy-configs firesim.firesim:FireSimRocketConfig
OpenJDK 64-Bit Server VM warning: Ignoring option MaxPermSize; support was removed in 8.0
Elaborating design...
L2 InclusiveCache Client Map:
	0 <= blkdev-tracker0
	1 <= serial-tl
	2 <= Core 0 DCache
	3 <= Core 0 ICache

Interrupt map (2 harts 2 interrupts):
  [1, 1] => frontend
  [2, 2] => uart_0

[error] java.nio.file.NoSuchFileException: /home/centos/chipyard_oss_210430/sims/firesim/sim/generated-src/f1/FireSim-FireSimRocketConfig-BaseF1Config/bootrom.rv64.img
[error] 	...
[error] 	at freechips.rocketchip.devices.tilelink.BootROM$.contents$lzycompute$1(BootROM.scala:75)
```

This seems strange but I guess it is probably related to why [`firesim.firesim.WithBootROM` didn't use the path at `chipyard/bootrom` and instead reached all the way down into testchipip to get the ROM img ](https://github.com/ucb-bar/chipyard/pull/872/files#diff-e4468b96461e4c7fcb103a17a6a01bc40296e9aeeb712d1ea4218d57b0dd3e80L28-L29)

Reply:
Oh, `GenerateSimFiles` is a separate app that the chipyard makefiles call before the generator runs?  @davidbiancolin, head's up, I'm going to add a call to the `chipyard.GenerateSimFiles` for the none simulator in place of this copying.

Reply:
@abejgonzalez , Do you guys have trouble with having changes happen to files copied by GenerateSimFiles not causing the copy to happen again?  Looking at the Chipyard makefiles, it seems that ones the `$(build_dir)/sim_files.f` is created, `GenerateSimFiles` will never be rerun, even if the resources it copied have changed.  Am I missing something?

Reply:
After thinking about this more, I feel like adding `GenerateSimFiles` usage to FireSim might not be the right way to go because:
1. it isn't being used by the FireSim simulation makefiles (MIDAS-level) and adding it only for the bootrom files might confuse or conflict with ML sim
2. It looks like it breaks the dependency information in the makefiles so that files used in `sim_files.f` aren't kept up to date inside of the `target_dir` (but maybe I'm missing some other part of the CY flow).

Reply:
Hmmm. I think we just missed that... Ideally, the `GenerateSimFiles` target depends on the items it potentially copies over but that isn't the case. I suppose no-one has gotten to that issue until now. Now I wonder if instead of using `GenerateSimFiles` we just do the same thing in the Makefiles (copy files over depending on if make is called from VCS or Verilator, and generating the `.f`). 

For now, I think it's fine to ignore it.